tools/CMakeLists.txt

```diff
@@ -19,6 +19,7 @@ add_subdirectory(terminal)
 add_subdirectory(recorder)
 add_subdirectory(fw-update)
 add_subdirectory(embed)
+add_subdirectory(dds/dds-sniffer)
```

Comment:
I say let's move the CMake flag condition check here, and ask nothing inside the DDS  CMakelist.txt file

Reply:
Even more than that:
It should have:
```
if (BUILD_WITH_DDS)
add_subdirectory(dds)
endif()
```
and then inside a new dds/CMakeLists.txt, it should have each DDS subproject.